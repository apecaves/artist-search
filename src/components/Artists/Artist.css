.Artist {
  color: black;
  text-decoration: none;
  &:focus,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
  &:hover {
    color: grey;
  }
}
