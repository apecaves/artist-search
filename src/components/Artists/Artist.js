import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Artist.css';

function Artist({ artist }) {
  return (
    <section >
      <Link className={styles.Artist} to={`/artist/${artist.name}/${artist.id}`}>{artist.name}</Link>
    </section>
  );
}

Artist.propTypes = {
  artist: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired
};

export default Artist;
