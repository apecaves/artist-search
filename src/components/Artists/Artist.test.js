import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

describe('Artist component', () => {
  it('renders Artist', () => {
    const wrapper = shallow(<Artist artist={{
      id: '', name: ''
    }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  