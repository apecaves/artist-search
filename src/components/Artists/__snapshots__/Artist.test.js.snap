// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders Artist 1`] = `
<section>
  <Link
    className="Artist"
    to="/artist//"
  />
</section>
`;
